app/scaffold-controllers/searchBarController.js

```diff
@@ -0,0 +1,97 @@
+define([
+    'bluebird',
+    'plugins/webViewPlugin'
+],
+function(
+    Promise,
+    WebViewPlugin
+) {
+    var SearchBarController = function(webView, layout, searchConfig) {
+        this.viewPlugin = webView;
+        this.layout = layout;
+        this.searchConfig = searchConfig;
+
+        // Track the state to show.hide keyboard on toggle
+        //
+        // Set to true because it will be hidden when added
+        // to the parent layout
+        this.isVisible = true;
```

Comment:
@tony-dinh the comment means that by default the search bar is visible - when it is added to the parent layout, it will be told to hide. (hide will change its logic if it thinks the search bar is already not visible) We could probably just do away with the comment.